src/main/java/gift/repository/JdbcProductRepository.java

```diff
@@ -43,30 +43,35 @@ private Product RowMapperProduct(ResultSet rs, int rowNum) throws SQLException {
 
     @Override
     public List<Product> findAllProducts() {
-        String sql = "SELECT * FROM products";
+        String sql = "SELECT * FROM product";
         return jdbcTemplate.query(sql, this::RowMapperProduct);
     }
 
     @Override
     public Optional<Product> findProductById(Long id) {
-        String sql = "SELECT * FROM products WHERE id = ?";
+        String sql = "SELECT * FROM product WHERE id = ?";
 
         try {
             Product product = jdbcTemplate.queryForObject(sql, this::RowMapperProduct, id);
-            return Optional.of(new Product(
-                    product.getId(),
-                    product.getName(),
-                    product.getPrice(),
-                    product.getImageUrl()
-            ));
+            return Optional.of(product);
         } catch (Exception e) {
             return Optional.empty();
         }
     }
 
     @Override
     public Optional<Product> updateProduct(Long id, Product product) {
-        return Optional.empty();
+
+        String sql = "SELECT * FROM product WHERE id = ?";
+
+        try {
+            jdbcTemplate.queryForObject(sql, this::RowMapperProduct, id);
```

Comment:
이 메서드는 변수를 선언하지 않는구요. 불필요한 쿼리로 보이네요-! 의도가 있으셨을까요?

Reply:
이건 무슨 말씀이신지 잘 모르겠습니다.
sql 변수를 말씀하시는거면 jdbcTemplate.queryForObject(sql, this::RowMapperProduct, id); 여기서 사용했는데 불필요한 쿼리라는게 무슨 뜻인가요?

Reply:
@hopencode SELECT 쿼리를 요청하는 이유가 있을까요-?

Reply:
@iphwade  그 다음 코드인 jdbcTemplate.update에서는 없는 id의 상품에 대한 업데이트를 시도해도 예외가 발생하지 않아서 없는 id의 상품인 경우 Optional.empty() 리턴을 위한 예외를 발생시키기 위해 SELECT 쿼리를 요청했습니다.
해당 쿼리는 상품이 없는 경우 예외가 발생합니다.
또 미션2 진행하면서 거기에 코드 리뷰 주신걸 반영했는데 미션1 코드에서도 반영한 코드로 다시 PR 올려드려야 하나요?

Reply:
@hopencode 
JdbcTemplate의 queryForObject가 데이터를 못 찾을 때 EmptyResultDataAccessException을 발생시키는 점을 이용해 '상품 없음' 케이스를 처리하려는 의도로 이해했습니다.

다만 현재 catch(Exception e) 구문은 너무 광범위해서, '단순히 상품이 없는 경우’'DB 서버 다운 같은 심각한 시스템 오류'를 구분하지 못하고 모두 Optional.empty()로 처리하게 됩니다.

이렇게 되면 DB 장애가 발생해도 시스템은 오류를 감지하지 못하고, 마치 상품이 없는 것처럼 조용히 넘어가 버릴 위험이 있습니다.
따라서 다음과 같이 역할을 명확히 분리해보는 것은 어떨까요? 

- try-catch 블록 : 예상 가능한 예외인 EmptyResultDataAccessException만 명시적으로 잡아 '상품 없음'이라는 정상적인 비즈니스 흐름을 처리합니다.
- 그 외 예외: DB 연결 실패나 타임아웃 같은 예측하지 못한 시스템 오류는 catch하지 않고 그대로 두어, 에러가 상위로 전파되거나 로그에 기록되게 해야 합니다. 이를 통해 시스템 장애를 빠르게 인지하고 대응할 수 있습니다.

Reply:
```suggestion
@Override
public Optional<Product> updateProduct(Long id, Product product) {
    String sql = "UPDATE product SET name = ?, price = ?, imageUrl = ? WHERE id = ?";
    
    int affectedRows = jdbcTemplate.update(sql, product.getName(), product.getPrice(), product.getImageUrl(), id);

    if (affectedRows > 0) {
        Product updatedProduct = new Product(id, product.getName(), product.getPrice(), product.getImageUrl());
        return Optional.of(updatedProduct);
    } else {
        // 혹은 에러를 발생시켜 상위 클래스에서 404 에러를 발생시킬 수도 있겠군요. 
        return Optional.empty();
    }
}
```

더 나아가 DB 접근을 1회로 줄이고, EmptyResultDataAccessException를 발생시키지 않도록 변경할 수도 있겠네요.
그 외 예측하지 못한 DB 장애와 같은 예외들은 상위로 전파되어 인지할 수 있겠군요.

Reply:
@hopencode 다시 코드 리뷰를 요청하지 않으셔도 됩니다. 변경하지 않으셔도 되구요-!
코멘트 읽고 이런 점도 있구나 정도로 넘어가셔도 충분해보입니다~ 
앞으로의 미션에 집중하는게 더 도움이 될 거라 생각해요!